developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,338 @@
+   with "Py" or "_Py", which is verified via ``make smelly``.
+
+One could imagine having more ``.h`` files, or no ``helper.c`` file if it is
+not needed. Here, we wanted to illustrate a simple example without making it
+too trivial.
+
+Make the CPython project compile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we have our files, we need to update the :cpy-file:`Makefile.pre.in` file.
+First, define the following the variables:
+
+.. code-block:: makefile
+
+   FOO_H = Modules/foo/foomodule.h
+   FOO_OBJS = Modules/foo/foomodule.o Modules/foo/helper.o
+
+and place them in the **Modules** section where other pre-defined objects live such
+as ``MODULE_OBJS`` and ``IO_OBJS``. Then, add the following rule in the section for
+**Special rules for object files**:
+
+.. code-block:: makefile
+
+   $(FOO_OBJS): $(FOO_H)
+
+and the following rule in the section for **Module dependencies and platform-specific files**:
+
+.. code-block:: makefile
+
+   MODULE_FOO_DEPS=$(srcdir)/Modules/foo/foomodule.h
+
+.. note::
+
+   The ``FOO_OBJS`` and ``FOO_H`` are not necessarily needed and the rule
+   ``$(FOO_OBJS): $(FOO_H)`` could be hard-coded. Using Makefile variables
+   is generally better if multiple files need to be compiled.
+
+Finally, we need to modify the configuration for Windows platforms:
+
+- Open :cpy-file:`PC/config.c` and add the prototype:
+
+  .. code-block:: c
+
+     extern PyObject* PyInit_foo(void);
+
+  and the entry ``{"foo", PyInit_foo}`` to ``_PyImport_Inittab``.
+
+- Open :cpy-file:`PCbuild/pythoncore.vcxproj` and add the following line to
+  the ``<ItemGroup>`` containing the other ``..\Modules\*.h`` files:
+
+  .. code-block:: xml
+
+     <ClInclude Include="..\Modules\foo\foomodule.h" />
+
+  In addition, add the following lines to the ``<ItemGroup>``
+  containing the the other ``..\Modules\*.c`` files:
+
+  .. code-block:: xml
+
+     <ClCompile Include="..\Modules\foo\foomodule.c" />
+     <ClCompile Include="..\Modules\foo\helper.c" />
+
+- Open :cpy-file:`PCbuild/pythoncore.vcxproj.filters` and add the following line to
+  the ``ItemGroup`` containing the the other ``..\Modules\*.h`` files:
+
+  .. code-block:: xml
+
+     <ClInclude Include="..\Modules\foo\foomodule.h">
+         <Filter>Modules\foo</Filter>
+     </ClInclude>
+
+  In addition, add the following lines to the ``ItemGroup`` containing
+  the the other ``..\Modules\*.c`` files:
+
+  .. code-block:: xml
+
+     <ClCompile Include="..\Modules\foo\foomodule.c">
+       <Filter>Modules\foo</Filter>
+     </ClCompile>
+     <ClCompile Include="..\Modules\foo\helper.c">
+       <Filter>Modules\foo</Filter>
+     </ClCompile>
+
+Observe that ``.h`` files use ``<ClInclude ...>`` whereas ``.c`` files
+use ``<ClCompile ...>`` tags.
+
+Compile the CPython project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that everything is in place, it remains to compile the project:
+
+.. code-block:: shell
+
+   $ make regen-configure
+   $ make regen-all
+   $ make regen-stdlib-module-names
```

Comment:
The copy button omits the dollar signs, but not when highlighting and manually copying.

```suggestion
   make regen-configure
   make regen-all
   make regen-stdlib-module-names
```